File: 1434-decrypt-string-from-alphabet-to-integer-mapping/1434-decrypt-string-from-alphabet-to-integer-mapping.cpp
class Solution {
public:
    string freqAlphabets(string s)
    {
     map<int, char> mp;

    //  string str[26];

    //  for(int i = 1; i <= 26; i++)
    //  {
    //      string res = to_string(i);
    //      str[i-1] = res;
    //  }

     for(char c = 'a',i = 1; c <= 'z'; c++,i++)
     {
         mp[i] = c;
     } 

    //  for(auto itr = mp.begin(); itr != mp.end(); itr++)
    //  {
    //      cout<<itr->first<<"->"<<itr->second<<"\n";
    //  }

    int i = 0;
    string res;
    while(i < s.length())
    {
        if(i + 2 < s.length() and s[i + 2] == '#')
        {
            string st = s.substr(i,2);
            int value = stoi(st);
            char c = mp[value];
            res += c;
            i = i + 3;
        }
        else
        {
            string str = s.substr(i,1);
            int value = stoi(str);
            char c = mp[value];
            res += c;
            i++;
        }
    }

        return res;
    }
};

File: 1434-decrypt-string-from-alphabet-to-integer-mapping/1434-decrypt-string-from-alphabet-to-integer-mapping.java
class Solution {
    public String freqAlphabets(String s) {
        
        char ch = 'a';
        int i = 1;
        
        HashMap<String, Character> mp = new HashMap<String, Character>();
        while(ch < 'j')
        {
            String str = String.valueOf(i);
            mp.put(str,ch);
            ch++;
            i++;
        }

        ch = 'j';
        i = 10;

        while(ch <= 'z')
        {
            String str = String.valueOf(i);
            mp.put(str,ch);
            ch++;
            i++;
        }
       

        for(Map.Entry<String, Character> e : mp.entrySet())
        {
            System.out.println(e);
        }

        StringBuilder res = new StringBuilder("");

        for(int j = s.length() - 1; j >= 0; j--)
        {
            if(s.charAt(j) != '#')
            {
                String str = String.valueOf(s.charAt(j));
                res.append(mp.get(str));
            }
            else
            {
                String str = s.substring(j-2,j);
                res.append(mp.get(str));
                j = j - 2;
            }
        }

        res.reverse();

        // System.out.println(cha);
        return res.toString();
    }
}